src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -844,48 +853,143 @@ QgsExpressionContext QgsLayoutItemLegend::createExpressionContext() const
   scope->addVariable( QgsExpressionContextScope::StaticVariable( QStringLiteral( "legend_filter_out_atlas" ), legendFilterOutAtlas(), true ) );
 
   context.appendScope( scope );
-
   return context;
 }
 
+QgsExpressionContext QgsLayoutItemLegend::updateExpressionContext()
```

Comment:
Update the the expression context that is now cached in this class. This is easier to handle it and pass it down.